src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            [this[i], this[j]] = [this[j], this[i]];
+          }
+        } else {
+          if (String(this[i]) > String(this[j])) {
+            [this[i], this[j]] = [this[j], this[i]];
+          }
+        }
+      }
+    }
+
+    return this;
```

Comment:
The sort2 method is modifying the array in place. This is consistent with the behavior of the native sort method. However, it is important to note that this side effect should be clearly documented.

Reply:
Where should it be documented?